vars/transportRequestUploadFile.groovy

```diff
@@ -0,0 +1,73 @@
+import com.sap.piper.GitUtils
+import groovy.transform.Field
+
+import com.sap.piper.ConfigurationMerger
+import com.sap.piper.cm.ChangeManagement
+import com.sap.piper.cm.ChangeManagementException
+
+import hudson.AbortException
+
+
+@Field def STEP_NAME = 'transportRequestUploadFile'
+
+@Field Set parameterKeys = [
+    'changeId',
+    'transportRequestId',
+    'applicationId',
+    'filePath',
+    'cmCredentialsId',
+    'cmEndpoint'
+  ]
+
+@Field Set generalConfigurationKeys = [
+    'cmCredentialsId',
+    'cmEndpoint'
+  ]
+
+def call(parameters = [:]) {
+
+    handlePipelineStepErrors (stepName: STEP_NAME, stepParameters: parameters) {
+
+        def script = parameters?.script ?: [commonPipelineEnvironment: commonPipelineEnvironment]
+
+        ChangeManagement cm = new ChangeManagement(script)
+
+        Map configuration = ConfigurationMerger.merge(parameters.script, STEP_NAME,
+                                                      parameters, parameterKeys,
+                                                      generalConfigurationKeys)
+
+        def changeId = configuration.changeId
```

Comment:
See above: changeId from commit messages

Reply:
I think that the `changeId` should be retrieve on a previous step, and it should be passed as parameter to this step.

Reply:
See other comment from me above. I can imagine well that this is the first cm related step. To my understanding it makes sense to retrieve the values also from a set of git commits. What is the benefit of having another step only retrieving the changeId, (maybe also the transport requestId) and using that step before any other cm releated steps? There are also known limitations wrt return values for declarative pipelines.